monai/metrics/confusion_matrix.py

```diff
@@ -92,27 +95,34 @@ def __call__(self, y_pred: torch.Tensor, y: torch.Tensor):
                 warnings.warn("As for classification task, compute_sample should be False.")
                 self.compute_sample = False
 
-        confusion_matrix = get_confusion_matrix(
+        return get_confusion_matrix(
             y_pred=y_pred,
             y=y,
             include_background=self.include_background,
         )
 
-        if self.compute_sample:
-            if isinstance(self.metric_name, str):
-                confusion_matrix = compute_confusion_matrix_metric(self.metric_name, confusion_matrix)
-                f, not_nans = do_metric_reduction(confusion_matrix, self.reduction)
-                return f, not_nans
-            if len(self.metric_name) < 1:
-                raise ValueError("the sequence should at least has on metric name.")
-            results = []
-            for metric_name in self.metric_name:
-                sub_confusion_matrix = compute_confusion_matrix_metric(metric_name, confusion_matrix)
+    def aggregate(self):
+        """
+        Execute reduction for the confusion matrix values.
+
+        """
+        super().aggregate()
```

Comment:
Currently, we only have the `sync` logic in `super().aggregate()`, so yes, it equals to call `get_buffer()` directly, but I think we may add more logic in the aggregate of base class, so I still call `super().aggregate()` here.

Reply:
perhaps the base class's `aggregate()` should do nothing, otherwise we have 3 ways to run `_sync`: `self._sync`, `self.aggregate`, `self.get_buffer`, but I couldn't find the corresponding use cases

Reply:
Hi @wyli ,

I think:
1. `self._sync()` is private function which is not supposed to be called by user.
2. `self.aggregate()` is the main function to compute values, it's better to automatically sync first, otherwise, users need to always call `sync` by themselves. And I can't find a use case that we should NOT sync before aggregation.
3. `self.get_buffer()` is a util function to get the metrics details or other use cases.

Thanks.

Reply:
agree that _sync wouldnt be used by the end user, but it could be used by subclasses

Reply:
Hi @wyli ,

Thanks for your suggestion, sounds make sense.
Let me try to update it.

Thanks.

Reply:
sure, I think the base class aggregate should be left not implemented, the subclasses should use _sync and get_buffer to implement it, that's what is implemented as well

Reply:
yes, I plan to update like that.
Thanks.

Reply:
Is there a use case that users just use this class to accumulate tensors across ranks?
```py
cum = Cumulative()
cum.add(...)
cum.add(...)
data = cum.get_buffer()
cum.reset()
```
If so, we should set `pass` to `aggregate()` instead of `implemented`.

Thanks.

Reply:
sure, I'm fine with either, depending on whether "do nothing" is considered a way of aggregation...

Reply:
updated to `implemented` in the latest commit.

Thanks.